src/matcha_ml/core/core.py

```diff
@@ -263,6 +301,26 @@ def provision(
         return matcha_state_service.fetch_resources_from_state_file()
 
 
-def stack_set(stack: str) -> str:
-    """Placeholder for core matcha stack set functionality."""
-    return stack
+def stack_set(stack_name: str) -> None:
+    """A function for updating the stack type in the local matcha.config.json file.
+
+    Args:
+        stack_name (str): the name of the type of stack to be specified in the config file.
+    """
+    if RemoteStateManager().is_state_provisioned():
+        raise RuntimeError(
```

Comment:
This error isn't captured by the call to this function in `cli.py`, the `try... except` there is capturing a `MatchaError`, perhaps raise that here instead of the `RunTimeError`?

Reply:
Good spot, I've instead added two types of error capture in cli.py so we can exit the typer neatly.